bentoml/_internal/server/__init__.py

```diff
@@ -230,9 +237,14 @@ def start_dev_api_server(
     reload: bool = False,
     reload_delay: t.Optional[float] = None,
     instance_id: t.Optional[int] = None,  # pylint: disable=unused-argument
+    custom_service_app_factory: t.Optional[str] = None,
 ):
     import uvicorn  # type: ignore
 
+    if custom_service_app_factory is not None:
+        klass = LazyType["ServiceAppFactory"](custom_service_app_factory).get_class()
```

Comment:
`LazyType` here acts as a typed `importlib`. The klass will be taken as `t.Type[ServiceAppFactory]` by type checkers